cellxgene_schema_cli/tests/test_schema_compliance.py

```diff
@@ -1519,6 +1531,64 @@ def test_obsm_shape_same_rows_and_columns(self):
             self.validator.adata.obsm["X_umap"] = three_row_array
             self.validator.validate_adata()
 
+    def test_obsm_size_zero(self):
+        """
+        size of obsm an key cannot be zero.
+        """
+        adata = self.validator.adata
+        adata.obsm["badsize"] = numpy.empty((2, 0))
+        self.validator.adata = self.save_and_read_adata(adata)
```

Comment:
why do we need to do this? Does doing the update in-memory ala `self.validator.adata.obsm["badsize"] = numpy.empty((2, 0))` not work for some reason? It works for other tests

Reply:
The problem is whatever anndata is doing when it saves. Otherwise there is no way to create numpy array with size 0.